src/open_inwoner/accounts/tests/test_auth_2fa_sms.py

```diff
@@ -0,0 +1,542 @@
+import datetime
+import time
+from unittest.mock import patch
+
+from django.conf import settings
+from django.core.signing import TimestampSigner
+from django.test import TestCase, override_settings
+from django.test.utils import freeze_time as dj_freeze_time
+from django.urls import reverse
+from django.utils import timezone
+from django.utils.http import urlencode
+from django.utils.translation import gettext as _
+
+from freezegun import freeze_time
+from timeline_logger.models import TimelineLog
+
+from open_inwoner.configurations.models import SiteConfiguration
+from open_inwoner.utils.logentry import LOG_ACTIONS
+from open_inwoner.utils.validators import format_phone_number
+
+from ..gateways import GatewayError
+from .factories import UserFactory
+
+signer = TimestampSigner()
+
+
+@override_settings(ROOT_URLCONF="open_inwoner.cms.tests.urls")
+class TestSMSVerificationLogin(TestCase):
+    def setUp(self):
+        self.user = UserFactory(email="ex@example.com", password="secret")
+
+        self.config = SiteConfiguration.get_solo()
+        self.config.login_2fa_sms = True
+        self.config.login_allow_registration = True
+        self.config.save()
+
+    @patch("open_inwoner.accounts.gateways.gateway.send")
+    def test_regular_login_with_2fa_disabled(self, mock_gateway_send):
+        self.config.login_2fa_sms = False
+        self.config.save()
+
+        response = self.client.get(reverse("login"))
+        self.assertFalse(mock_gateway_send.called)
+
+        response = self.client.post(
+            reverse("login"),
+            data={
+                "next": "",
+                "username": self.user.email,
+                "password": "secret",
+            },
+            follow=True,
+        )
+
+        self.assertNotContains(response, "Account verificatie")
+        self.assertFalse(mock_gateway_send.called)
+        self.assertEqual(mock_gateway_send.call_count, 0)
+        self.assertRedirects(response, reverse("pages-root"))
+
+    @patch("open_inwoner.accounts.gateways.gateway.send")
+    def test_login_with_valid_token_succeeds(self, mock_gateway_send):
+        response = self.client.get(reverse("login"))
+        self.assertFalse(mock_gateway_send.called)
+
+        response = self.client.post(
+            reverse("login"),
+            data={
+                "next": "",
+                "username": self.user.email,
+                "password": "secret",
+            },
+            follow=True,
+        )
+
+        self.assertContains(response, "Account verificatie")
+        self.assertTrue(mock_gateway_send.called)
+        self.assertEqual(mock_gateway_send.call_count, 1)
+        self.assertEqual(mock_gateway_send.call_args[1]["to"], self.user.phonenumber)
```

Comment:
You can assert on the mock itself: `mock.assert_called_once()` or `mock.assert_called_once_with(...)` (etc, see https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock)